pkg/k8sops/statefulset.go

```diff
@@ -254,3 +257,72 @@ func (k *K8sops) BuildM3DBCoordinatorPorts() []v1.ContainerPort {
 		},
 	}
 }
+
+func (k *K8sops) EnsureStatefulSets(cluster *myspec.M3DBCluster) (map[string]string, error) {
+	placementDetails := make(map[string]string) // key is pod name, value zone
+
+	// ensure zones are present in spec
+	if len(cluster.Spec.Zones) != 0 {
+		// TODO(PS) add zone distribution algo to range
+		// loop through zones
+		for zoneIndex, zoneReplicaCount := range []int32{1, 1, 1} {
+			statefulset, err := k.Kclient.AppsV1().
+				StatefulSets(cluster.GetNamespace()).
+				Get(k.StatefulsetName(cluster.GetObjectMeta().GetName(),
+					cluster.Spec.Zones[zoneIndex],
+					cluster.Spec.ReplicationFactor),
+					metav1.GetOptions{})
+			if len(statefulset.GetName()) == 0 {
+				// Create statefulsets
+				logrus.WithFields(logrus.Fields{
+					"zone":          cluster.Spec.Zones[zoneIndex],
+					"replicasCount": zoneReplicaCount}).
+					Info("building statefulset configuration")
+				statefulset, err = k.BuildStatefulset(
+					cluster.Spec,
+					cluster.ObjectMeta.Name,
+					cluster.ObjectMeta.Namespace,
+					cluster.Spec.Zones[zoneIndex],
+					&zoneReplicaCount,
+					cluster.Spec.ReplicationFactor,
+				)
+				if err != nil {
+					logrus.WithError(err).Error("failed to build statefulset")
+					return nil, err
+				}
+				if statefulset, err := k.Kclient.AppsV1().StatefulSets(cluster.GetNamespace()).Create(statefulset); err != nil {
+					logrus.WithError(err).WithField("statefulset", statefulset).Error("failed to create statefulset")
+				}
+				logrus.Info("statefulset created")
+
+				// Poll newly created stateful set and ensure all PODs are in ready state
+				err = wait.Poll(500*time.Millisecond, 60*time.Second, func() (bool, error) {
```

Comment:
nit: use configurable knobs for these values?

Reply:
Agreed, will move to the operator spec.